app/models/relationship.rb

```diff
@@ -177,11 +177,17 @@ def update_elasticsearch_parent(action = 'create_or_update')
   end
 
   def set_unmatched_field(value)
-    target = self.target
-    unless target.nil?
-      target.unmatched = value
-      target.skip_check_ability = true
-      target.save!
+    items = [self.target]
+    count = 0
+    # unmatch source when there is no other targets assign to same source
+    count = Relationship.where(source_id: self.source_id).where.not(target_id: self.target_id).count if value
+    items << self.source if count == 0
+    items.compact.each do |item|
```

Comment:
At a first glance I was concerned about this but then I realized that `items` will have at maximum 2 items.

Reply:
yes maximum two items but I added as array to avoid code climate issue